terraform/main.tf

```diff
@@ -0,0 +1,117 @@
+provider "aws" {
+  region = "us-east-1"
+  profile = "serverless"
+}
+
+data "aws_vpc" "default" {
+  vpc_id = "vpc-0d60ef77"
+}
+
+data "aws_subnet_ids" "all" {
+  vpc_id = data.aws_vpc.default.id
+}
+
+resource "aws_elasticache_cluster" "network-data" {
+  cluster_id           = "network-data"
+  engine               = "redis"
+  node_type            = "cache.t3.small"
+  num_cache_nodes      = 1
+  parameter_group_name = "default.redis6.x"
+  engine_version       = "6.x"
+  port                 = 6379
+}
+
+resource "aws_ecr_repository" "data-collector" {
+  name                 = "data-collector"
+  image_tag_mutability = "MUTABLE"
+}
+
+resource "aws_iam_role" "ec2_role_data_collector" {
+  name = "ec2_role_data_collector"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+
+  tags = {
+    project = "hello-world"
+  }
+}
+
+resource "aws_iam_instance_profile" "ec2_profile_data_collector" {
+  name = "ec2_profile_data_collector"
+  role = aws_iam_role.ec2_role_data_collector.name
+}
+
+resource "aws_iam_role_policy" "ec2_policy" {
+  name = "ec2_policy"
+  role = aws_iam_role.ec2_role_data_collector.id
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "ecr:GetAuthorizationToken",
+        "ecr:BatchGetImage",
+        "ecr:GetDownloadUrlForLayer"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+module "dev_ssh_sg" {
+  source = "terraform-aws-modules/security-group/aws"
```

Comment:
I think it's a good idea to have this kind of limitation by IP.
But this looks like something that should be in a global Terraform configuration in another repo.

What do you think?